packages/next/src/build/index.ts

```diff
@@ -2049,16 +2049,19 @@ export default async function build(
                             isInterceptionRouteAppPath(page)
                           if (appConfig.revalidate !== 0) {
                             const isDynamic = isDynamicRoute(page)
-                            const hasGenerateStaticParams =
-                              !!workerResult.prerenderRoutes?.length
+                            const generateStaticParamsIsDefined =
+                              !!workerResult.prerenderRoutes
+                            const generateStaticParamsHasItems =
+                              generateStaticParamsIsDefined &&
+                              workerResult.prerenderRoutes?.length
 
                             if (
                               config.output === 'export' &&
                               isDynamic &&
-                              !hasGenerateStaticParams
+                              !generateStaticParamsIsDefined
                             ) {
                               throw new Error(
-                                `Page "${page}" is missing "generateStaticParams()" so it cannot be used with "output: export" config.`
+                                `Page "${page}" is missing "generateStaticParams()" so it cannot be used with "output: export" config. Return an empty array if no pages should be rendered.`
```

Comment:
Extra context in this error message can point users in the right direction when running into this issue. This new description technically still isn't completely correct as we hit this error either if `generateStaticParams` or its result is undefined.